Simple patch to change default sorting by position to sorting by custom flag and price
Issue appeared in ticket: IPET-997

diff --git a/Model/Product/Type/Grouped.php b/Model/Product/Type/Grouped.php
index 24e4e49..124e639 100644
--- a/Model/Product/Type/Grouped.php
+++ b/Model/Product/Type/Grouped.php
@@ -215,7 +215,13 @@ class Grouped extends \Magento\Catalog\Model\Product\Type\AbstractType
                 $product
             )->addAttributeToSelect(
                 ['name', 'price', 'special_price', 'special_from_date', 'special_to_date', 'tax_class_id', 'image']
-            )->addFilterByRequiredOptions()->setPositionOrder()->addStoreFilter(
+            )->addFilterByRequiredOptions();
+            // BEGIN PATCH
+            // Change default sorting by position to sorting by custom flag and price
+            $collection->addAttributeToSort('skip_price_calculation_in_grouped_product', \Magento\Framework\Api\SortOrder::SORT_ASC)
+                ->addAttributeToSort('price', \Magento\Framework\Api\SortOrder::SORT_ASC);
+            // END PATCH
+            $collection->addStoreFilter(
                 $this->getStoreFilter($product)
             )->addAttributeToFilter(
                 'status',
